repose-aggregator/components/filters/herp-filter/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala

```diff
@@ -121,8 +123,17 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
       else stripHeaderParams(allProjectIds.maxBy(getQuality))
     }
 
-    val projectIds = httpServletRequest.getHeaders(OpenStackServiceHeader.TENANT_ID.toString).asScala
-      .++(httpServletRequest.getHeaders(X_PROJECT_ID).asScala).toTraversable
+    def getSplitHeaders(headerInteractor: HeaderInteractor, headers: List[String]): Traversable[String] = {
+      headers.foldLeft(Traversable[String]()) { (accumulator, current) =>
+        accumulator.++(headerInteractor.getSplittableHeaders(current).asScala)
+      }
+    }
+
+    val tenantProjectHeaders = List(OpenStackServiceHeader.TENANT_ID.toString, X_PROJECT_ID)
+    val projectIds = getSplitHeaders(new HttpServletRequestWrapper(httpServletRequest), tenantProjectHeaders) match {
+      case x if x.nonEmpty => x
+      case _ => getSplitHeaders(httpServletResponse, tenantProjectHeaders)
+    }
```

Comment:
That's true. In this case, I'm pretty sure the pattern match is actually compiled to the same bytecode as a simple conditional would have been. If you really wanted, you could put both lines into a code block to reduce the scope of the variable, but as you said, there probably isn't a reason to. Also, that gets messy again, at which point using pattern matching may have been clearer choice anyway.

If you like the pattern match, you can totally keep using it. I have weird preferences sometimes.